axes/handlers/database.py

```diff
@@ -152,18 +152,21 @@ def user_login_failed(
             else:
                 separator = "\n---------\n"
 
+                failures_since_start_prev = attempt.failures_since_start
+
                 attempt.get_data = Concat("get_data", Value(separator + get_data))
                 attempt.post_data = Concat("post_data", Value(separator + post_data))
                 attempt.http_accept = request.axes_http_accept
                 attempt.path_info = request.axes_path_info
-                attempt.failures_since_start += 1
+                attempt.failures_since_start = F("failures_since_start") + 1
                 attempt.attempt_time = request.axes_attempt_time
                 attempt.save()
 
                 log.warning(
                     "AXES: Repeated login failure by %s. Count = %d of %d. Updated existing record in the database.",
                     client_str,
-                    attempt.failures_since_start,
+                    # can be different from the actual value in the DB
```

Comment:
This breaks the idea of atomicity, really. Would it be more useful to just deprecate the count from this log message, if we can't reliably present it?

Reply:
Yes, I think that's an option. I'll remove it.